tests/test_objective_funs.py

```diff
@@ -353,3 +355,45 @@ def test_target_profiles():
     np.testing.assert_allclose(
         objc.target, current(objc.grid.nodes[objc.grid.unique_rho_idx])
     )
+
+
+@pytest.mark.unit
+def test_plasma_vessel_distance():
+    """Test calculation of min distance from plasma to vessel."""
+    R0 = 10.0
+    a_p = 1.0
+    a_s = 2.0
+    # default eq has R0=10, a=1
+    eq = Equilibrium(M=3, N=2)
+    # surface with same R0, a=2, so true d=1 for all pts
+    surface = FourierRZToroidalSurface(
+        R_lmn=[R0, a_s], Z_lmn=[-a_s], modes_R=[[0, 0], [1, 0]], modes_Z=[[-1, 0]]
+    )
+    # For equally spaced grids, should get true d=1
+    surf_grid = LinearGrid(M=5, N=6)
+    plas_grid = LinearGrid(M=5, N=6)
+    obj = PlasmaVesselDistance(
+        eq=eq, plasma_grid=plas_grid, surface_grid=surf_grid, surface=surface
+    )
+    d = obj.compute(*obj.xs(eq))
+    np.testing.assert_allclose(d, a_s - a_p)
+
+    # for unequal M, should have error of order M_spacing*a_p
```

Comment:
can you explain this geometry?

Reply:
The geometry is the same for all test cases. Plasma is a circular tokamak of R0=10, a=1, and the boundary surface is the same with a=2, so the true distance is 1 for all points. However, if the grids are different, the pointwise distances that we compute won't be the true min, but will have an error on the order of the arclength between points, which is the grid spacing * minor radius (basically the triangle inequality)